public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -785,13 +785,31 @@ export class ElasticDatasource
           fieldNameParts.pop();
         }
 
+        // Process runtime fields
+        function getRuntimeFields(mapping: { runtime: Record<string, { type: string }> }) {
```

Comment:
nit: This is kinda hard to argue since we are doing the same thing above, but I think we shouldn't have side effects on the parent's scope with `fields`. Can we pass fields in to make it more explicit that we are editing fields?